internal/services/api/post_config_secrets.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+// Copyright (C) 2024 by Posit Software, PBC.
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/posit-dev/publisher/internal/config"
+	"github.com/posit-dev/publisher/internal/logging"
+	"github.com/posit-dev/publisher/internal/types"
+	"github.com/posit-dev/publisher/internal/util"
+)
+
+const (
+	secretActionAdd    = "add"
+	secretActionRemove = "remove"
+)
+
+type postConfigSecretsRequest struct {
+	Action string `json:"action"`
+	Secret string `json:"secret"`
+}
+
+func applySecretAction(cfg *config.Config, action string, secret string) error {
```

Comment:
I know that this new endpoint is highly based on `post_config_files.go`. Sharing here some suggestions though, it is a good practice to keep API handlers lean and only consume service providers or instances that handle the main process. Meaning, an ideal flow would be
- Enpdoint handler gets called with some payload
- Payload is validated
- Payload is handled over to some service which is the responsible for the main action
- Endpoint handler returns the result

With that said, and if you are up to, you could create a new method into the `Config` struct responsible for all of these operations. IMO the config package is better suited to house this func. So your endpoint down below would look something like:
```go
    err = cfg.ApplySecretAction(b.Action, b.Secret)
    if err != nil {
        BadRequest(w, req, log, err)
        return
    }
```

Also, actions could be constants from the config package
```go
// config.go
const SecretActionAdd = "add"
const SecretActionRemove = "remove"
```